src/internal/components/dropdown/dropdown-fit-handler.ts

```diff
@@ -116,22 +121,46 @@ export const getInteriorAvailableSpace = (
   );
 };
 
-export const getDropdownPosition = (
-  trigger: HTMLElement,
-  dropdown: HTMLElement,
-  overflowParents: ReadonlyArray<Dimensions>,
-  minWidth?: number,
+export const getDropdownPosition = ({
+  triggerElement,
+  dropdownElement,
+  overflowParents,
+  minWidth: desiredMinWidth,
   preferCenter = false,
   stretchWidth = false,
   stretchHeight = false,
-  isMobile?: boolean
-): DropdownPosition => {
-  const availableSpace = getAvailableSpace(trigger, dropdown, overflowParents, stretchWidth, stretchHeight, isMobile);
-  const triggerWidth = trigger.getBoundingClientRect().width;
-  minWidth = minWidth ? Math.min(triggerWidth, minWidth) : triggerWidth;
-  const requiredWidth = dropdown.getBoundingClientRect().width;
-  // dropdown should not be smaller than the trigger
-  const idealWidth = Math.max(requiredWidth, minWidth);
+  isMobile = false,
+  stretchBeyondTriggerWidth = false,
+}: {
+  triggerElement: HTMLElement;
+  dropdownElement: HTMLElement;
+  overflowParents: ReadonlyArray<Dimensions>;
+  minWidth?: number;
+  preferCenter?: boolean;
+  stretchWidth?: boolean;
+  stretchHeight?: boolean;
+  isMobile?: boolean;
+  stretchBeyondTriggerWidth?: boolean;
```

Comment:
Reminder about this tech debt item: AWSUI-13040

This code is overloaded with flags which may contradict each other. There is an opportunity to split this code into separate "position strategies" for different use-cases.

Not a blocker, but I wanted to raise your attention to this

Reply:
Yeah, totally valid point. Unfortunately, I didn't really have the bandwidth to take that on.